### ISSUE TITLE
Stopping the VoicePipelineAgent from producing a response after a tool call

### ISSUE THREAD

**Issue by S4mpl3r** [livekit/agents#1060]
It would be amazing if we could control how the agent behaves when it calls tools. Currently, it first calls the tool and then calls the model immediately again to generate a response. I want to be able to change this behavior.

It is possible to prompt the underlying model to generate a text response AND a tool call in the same chat completion response, to tell things like "I'm getting the weather information for x" which can act as a filler until we get the tool call result (tested this on GPT-4o-mini). 
I want the agent to handle both the text message and the tool call (and not generate any automatic responses after calling the tool). Right now when I prompt the model to both generate a text message and tool call in the same response, it seems like the agent just cares about the text (it synthesizes speech) but completely ignores the tool call.
In a nutshell, I would like to be able to:
- Have control over whether the agent generates a completion after a tool call automatically or not
- Handle cases where the llm produces a text message and tool call in the same chat completion response

And just to be clear, I'm talking about the VoicePipelineAgent here.

**Comment by davidzhao** [comment]
The tool call is returning data to the model in order to generate a response. You can return `False` from function calls to prevent it to generate new responses

**Comment by S4mpl3r** [comment]
> The tool call is returning data to the model in order to generate a response. You can return `False` from function calls to prevent it to generate new responses

Thank you for the response. I tried what you said, but I got the following error:
```
openai.BadRequestError: Error code: 400 - {'error': {'message': "Invalid value for 'content': expected a string, got null.", 'type': 'invalid_request_error', 'param': 'messages.[4].content', 'code': None}}
```
I created a get_weather tool and inside the body of that function I just returned False, and I got the above error.

**Comment by theomonnom** [comment]
Hey, it isn't False but None. We're going to document that better

**Comment by S4mpl3r** [comment]
I tried it again (returning None instead of False), and it seems to be working as intended now. However, it bugs out after the tool call (nothing happens after the `received first TTS frame` event). As you've closed this issue, should I open another issue for that?

**Comment by davidzhao** [comment]
reopened to track it here.

**Comment by jayeshp19** [comment]
Hey @S4mpl3r Can you share your code? I'd love to look into it

**Comment by S4mpl3r** [comment]
Update: I've been trying it out again today, and it's working, I don't see the issue with it being stuck and not responding anymore. It might have been a false positive on my side. I'll share the test code that I've been using here in case others want to take a look:
```
import logging
from typing import Annotated

import aiohttp
from dotenv import load_dotenv
from livekit.agents import (
    AutoSubscribe,
    JobContext,
    JobProcess,
    WorkerOptions,
    cli,
    llm,
)
from livekit.agents.pipeline import VoicePipelineAgent
from livekit.plugins import deepgram, openai, silero

load_dotenv(dotenv_path=".env.local", override=True)
logger = logging.getLogger("voice-agent")


class AssistantFnc(llm.FunctionContext):
    @llm.ai_callable()
    async def get_weather(
        self,
        location: Annotated[
            str, llm.TypeInfo(description="The location to get the weather for")
        ],
    ):
        """Called when the user asks about the weather. This function will return the weather for the given location."""
        logger.info(f"getting weather for {location}")
        return None
        # url = f"https://wttr.in/{location}?format=%C+%t"
        # async with aiohttp.ClientSession() as session:
        #     async with session.get(url) as response:
        #         if response.status == 200:
        #             weather_data = await response.text()
        #             # response from the function call is returned to the LLM
        #             # as a tool response. The LLM's response will include this data
        #             return f"The weather in {location} is {weather_data}."
        #         else:
        #             raise f"Failed to get weather data, status code: {response.status}"


def prewarm(proc: JobProcess):
    proc.userdata["vad"] = silero.VAD.load()


def before_llm_cb(agent: VoicePipelineAgent, chat_ctx: llm.ChatContext):
    logger.info(f"before_llm_cb: {chat_ctx}")


async def entrypoint(ctx: JobContext):
    initial_ctx = llm.ChatContext().append(
        role="system",
        text=(
            # "You are a helpful assistant. When you want to call the get_weather function, you must respond with an approprate message to the user first, then call the function. BOTH IN THE SAME MESSAGE."
            "You are a helpful assistant. Talk to the user like you would in a normal conversation."
        ),
    )

    logger.info(f"connecting to room {ctx.room.name}")
    await ctx.connect(auto_subscribe=AutoSubscribe.AUDIO_ONLY)

    # Wait for the first participant to connect
    participant = await ctx.wait_for_participant()
    logger.info(f"starting voice assistant for participant {participant.identity}")

    assistant = VoicePipelineAgent(
        vad=ctx.proc.userdata["vad"],
        stt=deepgram.STT(),
        llm=openai.LLM(model="gpt-4o-mini"),
        tts=openai.TTS(),
        chat_ctx=initial_ctx,
        fnc_ctx=AssistantFnc(),
    )

    assistant.start(ctx.room, participant)

    await assistant.say("Hey, how are you doing?", allow_interruptions=True)


if __name__ == "__main__":
    cli.run_app(
        WorkerOptions(
            entrypoint_fnc=entrypoint,
            prewarm_fnc=prewarm,
        ),
    )
```
@davidzhao @jayeshp19

**Comment by davidzhao** [comment]
Thanks for the update @S4mpl3r, closing the issue this time.
